src/wp-includes/block-template-utils.php

```diff
@@ -608,10 +608,9 @@ function _build_block_template_result_from_file( $template_file, $template_type
 
 	$before_block_visitor = '_inject_theme_attribute_in_template_part_block';
 	$after_block_visitor  = null;
-	$hooked_blocks        = get_hooked_blocks();
-	if ( ! empty( $hooked_blocks ) || has_filter( 'hooked_block_types' ) ) {
-		$before_block_visitor = make_before_block_visitor( $hooked_blocks, $template, 'insert_hooked_blocks_and_set_ignored_hooked_blocks_metadata' );
-		$after_block_visitor  = make_after_block_visitor( $hooked_blocks, $template, 'insert_hooked_blocks_and_set_ignored_hooked_blocks_metadata' );
+	if ( maybe_has_hooked_blocks() ) {
+		$before_block_visitor = make_before_block_visitor( $template, 'insert_hooked_blocks_and_set_ignored_hooked_blocks_metadata' );
+		$after_block_visitor  = make_after_block_visitor( $template, 'insert_hooked_blocks_and_set_ignored_hooked_blocks_metadata' );
 	}
 	$blocks            = parse_blocks( $template->content );
 	$template->content = traverse_and_serialize_blocks( $blocks, $before_block_visitor, $after_block_visitor );
```

Comment:
`_build_block_template_result_from_post` has these lines inside the conditional above it. Is there a reason we aren't doing the same here?

Reply:
If memory serves, it's because we need to inject the `theme` attribute into template parts (and set it to the currently active theme) when first loading a template part from a file. For better or worse, that logic [ended up](https://github.com/WordPress/wordpress-develop/blob/c1c8d30398f4b8f7971b8e6ee6f793aca5f97be0/src/wp-includes/blocks.php#L1218) in `make_before_block_visitor`.

Now the need for that attribute is of course unrelated to the presence of hooked blocks -- it needs to be injected unconditionally 😬 

(I'd be curious if we have test coverage at that level, i.e. if a test would start failing if we moved these lines into the conditional.)